documentation/development-docs/examples/new-cmdlet-example.md

```diff
@@ -0,0 +1,180 @@
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = CreateParameterSet)]
+        public Type1 Property1 { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = CreateParameterSet)]
+        public Type2 Property2 { get; set; }
+
+        // Excluding other property parameters
+
+        public override void ExecuteCmdlet()
+        {
+            TopLevelResource existingResource = null;
+            try
+            {
+                existingResource = this.MySDKClient.TopLevelResource.Get(this.ResourceGroupName, this.Name);
+            }
+            catch
+            {
+                existingResource = null;
+            }
+
+            if (existingResource != null)
+            {
+                throw new Exception(string.Format("A TopLevelResource with name '{0}' in resource group '{1}' already exists. Please use Set/Update-AzTopLevelResource to update an existing TopLevelResource.", this.Name, this.ResourceGroupName));
+            }
+
+            existingResource = new TopLevelResource()
+            {
+                Name = this.Name,
+                ResourceGroupName = this.ResourceGroupName,
+                Property1 = this.Property1,
+                Property2 = this.Property2,
+                ...
+            }
+
+            if (this.ShouldProcess(this.Name, string.Format("Creating a new TopLevelResource in resource group '{0}' with name '{1}'.", this.ResourceGroupName, this.Name))
+            {
+                var result = new PSTopLevelResource(this.MySDKClient.TopLevelResource.CreateOrUpdate(existingResource));
+                WriteObject(result);
+            }
+        }
+    }
+}
+```
+
+### Child resource
+
+All child resources should have a `New-*` cmdlet that allows users to create a child resource with given properties. Properties that are required by the API should be mandatory parameters, and in the case where different combinations of properties are needed depending on a provided value (_e.g._, Windows and Linux VMs have different properties), multiple parameter sets should be used. This cmdlet should implement `SupportsShouldProcess` to allow users to provide the `-WhatIf` parameter and see what the result of executing the cmdlet is without the resource actually being created.
+
+_Note_: for long-running operations (~15s or longer), it is advised to add the [`-AsJob`](../design-guidelines/cmdlet-best-practices.md#asjob) to your cmdlet.
+
+#### Parameter sets
+
+To enable the above scenario, the cmdlet will need two parameter sets:
+
+```
+New-AzChildResource -ResourceGroupName <String> -TopLevelResourceName <String> -Name <String> [-Property1 <Type1>] [-Property2 <Type2>] ... [-WhatIf] [-Confirm]
+
+New-AzChildResource -TopLevelResourceObject <PSTopLevelResource> -Name <String> [-Property1 <Type1>] [-Property2 <Type2>] ... [-WhatIf] [-Confirm]
```

Comment:
Piping by parent resource id?

Reply:
@maddieclayton same comment as above -- this isn't a scenario we support